src/Subscriber/PaymentSubscriber.php

```diff
@@ -315,45 +318,50 @@ public function onCheckoutConfirmLoaded(PageLoadedEvent $event)
+        // Get Adyen /paymentMethods response
         $adyenPaymentMethods = $this->paymentMethodsService->getPaymentMethods($salesChannelContext);
 
-        foreach ($originalPaymentMethods as $paymentMethodEntity) {
-            $pmHandlerIdentifier = $paymentMethodEntity->getHandlerIdentifier();
+        // If the /paymentMethods response returns empty, remove all Adyen payment methods from the list and return
+        if (empty($adyenPaymentMethods['paymentMethods'])) {
+            return $originalPaymentMethods->filter(function (PaymentMethodEntity $item) use ($adyenPluginId) {
+                return $item->getPluginId() !== $adyenPluginId;
+            });
+        }
 
+        foreach ($originalPaymentMethods as $paymentMethodEntity) {
             //If this is an Adyen PM installed it will only be enabled if it's present in the /paymentMethods response
-            if (strpos($paymentMethodEntity->getFormattedHandlerIdentifier(), 'adyen') !== false) {
+            /** @var PaymentMethodEntity $paymentMethodEntity */
+            if ($paymentMethodEntity->getPluginId() === $adyenPluginId) {
+                $pmHandlerIdentifier = $paymentMethodEntity->getHandlerIdentifier();
                 $pmCode = $pmHandlerIdentifier::getPaymentMethodCode();
-                // In case the paymentMethods response has no payment methods, remove it from the list
-                if (empty($adyenPaymentMethods)) {
-                    $originalPaymentMethods->remove($paymentMethodEntity->getId());
-                    continue;
-                }
 
-                $methodFoundInResponse = array_filter(
-                    $adyenPaymentMethods['paymentMethods'],
-                    function ($value) use ($pmCode) {
-                        return $value['type'] == $pmCode;
+                if ($pmCode == OneClickPaymentMethodHandler::getPaymentMethodCode()) {
+                    if (empty($adyenPaymentMethods[OneClickPaymentMethodHandler::getPaymentMethodCode()])) {
+                        $originalPaymentMethods->remove($paymentMethodEntity->getId());
                     }
-                );
-
-                $shopwareMethodIsOneClick = $pmCode == OneClickPaymentMethodHandler::getPaymentMethodCode();
-                $oneClickInResponse = !empty(
-                    $adyenPaymentMethods[OneClickPaymentMethodHandler::getPaymentMethodCode()]
-                );
+                } else {
+                    $paymentMethodFoundInResponse = array_filter(
+                        $adyenPaymentMethods['paymentMethods'],
+                        function ($value) use ($pmCode) {
+                            return $value['type'] == $pmCode;
+                        }
+                    );
 
-                //Remove the PM if it isn't in the paymentMethods response
-                //and if it is OneClick while not present in the response
-                if (empty($methodFoundInResponse) && ($shopwareMethodIsOneClick && !$oneClickInResponse)) {
```

Comment:
This logic only removed onclick when oneclick is not in the response, but ignored all other PMs where `$methodFoundInResponse` was empty